### PR TITLE
Suggest to run Docker for Mac if it isn't running

### PR DESCRIPTION
**Pull request by bfirsh**
Instead of suggesting docker-machine.

Signed-off-by: Ben Firshman <ben@firshman.co.uk>

### CODE REVIEW

File: compose/cli/errors.py [hunk 1]
```diff
@@ -55,6 +56,8 @@ def handle_connection_errors(client):
             if is_ubuntu():
                 exit_with_error(docker_not_found_ubuntu)
             exit_with_error(docker_not_found_generic)
+        if is_docker_for_mac_installed():
+            exit_with_error(conn_error_docker_for_mac)
```
Comment (dnephin) [inline compose/cli/errors.py]: Should this go above the first `if which docker ` line? Or does docker for mac not include a docker binary in the path?

Comment (bfirsh) [inline compose/cli/errors.py]: It does include a docker binary in the path, so putting it after produces the correct behaviour:

 * If Docker for Mac is not installed, there is no binary in the path, so it'll show an error telling you to install it.
 * If Docker for Mac is installed and not running, it'll show a message telling you to start it.

Comment (bfirsh) [inline compose/cli/errors.py]: Docker for Mac also puts Compose in the path, so first scenario is unlikely. ;)

**Comment by dnephin** [comment]
Build failures are real, otherwise LGTM

**Comment by bfirsh** [comment]
fixed

**Comment by dnephin** [comment]
Looks like RC3 is still breaking the test suite.

LGTM (test failures are not related)

**Comment by aanand** [comment]
LGTM

**Comment by aanand** [comment]
Those failures are actually on 1.11 and are my fault - fix is in https://github.com/docker/compose/pull/3691.

**Pull request merged**